tests/common/include/tests/common/tools/event_writer.hpp

```diff
@@ -22,10 +22,14 @@ struct event_writer : actor {
     using scalar_type = typename transform3_t::scalar_type;
 
     struct state {
-        state(std::size_t event_id, smearer_t smearer)
-            : m_particle_writer(get_event_filename(event_id, "-particles.csv")),
-              m_hit_writer(get_event_filename(event_id, "-hits.csv")),
-              m_meas_writer(get_event_filename(event_id, "-measurements.csv")),
+        state(std::size_t event_id, const std::string directory,
```

Comment:
If you put the ```directory``` variable to the end of the method signature, you can set the empty string as default argument and don't have to change so much code later on. Same for the ```simulator```

Reply:
I fixed `simulator` to have the default argument. (But not for `event writer`)